### DISCUSSION TITLE
Webdriver IO opening multiple chrome instances in parallel even though I haven't specified the tests to run in parallel

### DISCUSSION THREAD

**Discussion by afsal-backer** [webdriverio/webdriverio#6672] [Q&A] [answered]
When I run wdio test, I see multiple chrome instances opening up parallelly even though I haven't specified the tests to run in parallel.

My **package.json** contains:

```
  "scripts": {
    "node-inspector": "node-inspector --debug-port 5859 --no-preload",
    "test": "node node_modules/.bin/wdio ./test/config/suite.jasmine.conf.js",
    "allure-report": "node_modules/.bin/allure generate ./test/reports/allure-results && allure report",
    "junit-report": "node_modules/.bin/junit-viewer --results=./test/reports/junit-results --save=junit-results.html && open junit-results.html",
    "generate-report": "./node_modules/.bin/allure generate  --clean && ./node_modules/.bin/allure open"
  },

  "devDependencies": {
    "@wdio/allure-reporter": "^5.22.0",
    "@wdio/cli": "^5.22.4",
    "@wdio/devtools-service": "^6.1.5",
    "@wdio/local-runner": "^5.3.5",
    "chai": "^4.2.0",
    "grunt": "^1.0.3",
    "grunt-cli": "^1.3.2",
    "grunt-jasmine": "^0.1.0",
    "grunt-webdriver": "^3.0.0",
    "rimraf": "^3.0.2",
    "selenium-standalone": "^6.15.4",
    "wdio-allure-reporter": "^0.8.3",
    "wdio-testrail-reporter": "^1.1.5",
    "webdriverio": "^6.1.7"
  },
...
```

my test>config>suite.jasmine.conf.js contains:

```
specs: [
     './test/specs/Lightning/Lead/lightning_leadRouting.spec.js' 
],

 maxInstances: 1,
 services: ['devtools', 'chromedriver','selenium-standalone'],
 framework: 'jasmine',
...

```

I am running the test with the command: ```npm test```

This opens up many browsers in parallel even though there is only one test to execute. Any help will be much appreciated.

**Comment by christian-bromann** [comment, marked as answer]
@gm4afsal please provide a reproducible example. Also you are using a incompatible WebdriverIO versions (v5 vs v6). I recommend to update to the latest version.

**Reply by afsal-backer** [reply]
I was able to solve the multiple chrome instances issue by tweaking:

   baseUrl: 'test.salesforce.com',
    waitforTimeout: 10000,   
    connectionRetryTimeout: 90000,  
    connectionRetryCount: 3,          **// to connectionRetryCount: 1,**
    wdioRetries: 3,   **// to  wdioRetries: 1**
    
    
   Now it only opens one browser for test but fails with the below error.
   
   
   Execution of 1 spec files started at 2021-04-08T04:49:54.305Z

```
[0-0] RUNNING in chrome - /test/specs/Lightning/Lead/lightning_leadRouting.spec.js
[0-0] Error: Failed to create session.
unknown error: Chrome failed to start: exited normally
  (unknown error: unable to discover open pages)
  (The process started from chrome location /Applications/Google Chrome.app/Contents/MacOS/Google Chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
  (Driver info: chromedriver=2.43.600229 (3fae4d0cda5334b4f533bede5a4787f7b832d052),platform=Mac OS X 10.15.7 x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 7.64 seconds
Build info: version: '3.141.5', revision: 'd54ebd709a', time: '2018-11-06T11:58:47'
System info: host: 'C02F5BL5MD6M', ip: 'fe80:0:0:0:fc:35fc:1ad3:1900%en0', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.15.7', java.version: '16'
Driver info: driver.version: unknown
[0-0] FAILED in chrome - /test/specs/Lightning/Lead/lightning_leadRouting.spec.js

Spec Files:      0 passed, 1 failed, 1 total (100% completed) in 00:00:30 

npm ERR! Test failed.  See above for more details.
```
   
 So I guess multiple browsers were opened up because of the above error and connectionRetryCount & wdioRetries : 3 
 
 In pursuit of solving the session creation error now.

**Reply by christian-bromann** [reply]
It seems that Chrome crashes on your system, did you install the right version of it, e.g. 64 vs 32 bit?

**Reply by afsal-backer** [reply]
Yes, it's fixed now.
Upgraded Wdio and Jasmine dependencies to the latest versions, added
 "chromedriver": "^89.0.0",
 "wdio-chromedriver-service": "^7.0.0" and then did an 'npm install' from the root directory.
